src/streamlink/plugins/cbsnews.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import print_function
+
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.stream import HLSStream
+from streamlink.utils import parse_json
+
+
+class CBSNews(Plugin):
+    url_re = re.compile(r"https://www\.cbsnews\.com/live/")
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return cls.url_re.match(url) is not None
+
+    def _get_streams(self):
+        res = self.session.http.get(self.url)
+        _re_json = re.compile(r"CBSNEWS.defaultPayload = (\{.*)")
+        m = _re_json.search(res.text)
+        if m:
+            data = parse_json(m.group(1))
+            for i in data['items']:
+                for s in HLSStream.parse_variant_playlist(self.session, i['video']).items():
+                    yield s
```

Comment:
The HLS URL should ideally be extracted via a validation schema.

```suggestion
_re_default_payload = re.compile(r"CBSNEWS.defaultPayload = (\{.*)")

_schema_items = validate.Schema(
    _re_default_payload.search,
    validate.any(None, validate.all(
        validate.get(1),
        validate.transform(parse_json),
        {"items": [validate.all({
            "video": validate.url(),
            "format": "application/x-mpegURL"
        }, validate.get("video"))]},
        validate.get("items")
    ))
)

def _get_streams(self):
    items = self.session.get(self.url, schema=self._schema_items)
    if items:
        for hls_url in items:
            for s in HLSStream.parse_variant_playlist(self.session, hls_url).items():
                yield s
```

Also please don't mix quotation styles.

Reply:
I did not feel it useful or necessary to use the validation methods.  
Bit nit-picky on the quotes, but OK.